controllers/auth.js

```diff
@@ -66,7 +70,51 @@ const signout = (req, res) => {
   });
 };
 
+/**
+ * Stores user-device data inside the DB for mobile auth
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const storeUserDeviceInfo = async (req, res) => {
+  try {
+    const userJson = {
+      user_id: req.body.user_id,
+      device_info: req.body.device_info,
+      device_id: req.body.device_id,
+      authorization_status: "NOT_INIT",
+    };
+    const userData = await QrCodeAuthModel.getUserAuthStatus(req.body.user_id);
+
+    const data = userData.docs.length ? userData.docs[0].data() : null;
+
+    if (data?.authorization_status) {
+      return res.status(409).json({
+        message: USER_DATA_ALREADY_PRESENT,
+      });
+    }
+
+    const userInfo = await QrCodeAuthModel.storeUserDeviceInfo(userJson);
+
+    if (!userInfo) {
+      return res.status(400).json({
+        message: BAD_REQUEST,
+      });
+    }
+
+    return res.status(201).json({
+      ...userInfo,
+      message: DATA_ADDED_SUCCESSFULLY,
+    });
+  } catch (err) {
+    logger.error(`Error while storing user device info : ${err}`);
+    return res.boom.badImplementation(SOMETHING_WENT_WRONG);
+  }
```

Comment:
Nit: Catch specific errors instead of a generic catch all. This helps with better error handling.